src/box/box.cc

```diff
@@ -153,7 +153,7 @@ request_rebind_to_primary_key(struct request *request, struct space *space,
 {
 	Index *primary = index_find_xc(space, 0);
 	uint32_t key_len;
-	char *key = tuple_extract_key(found_tuple,
+	char *key = primary->index_def->key_def.tuple_extract_key(found_tuple,
```

Comment:
Please keep tuple_extract_key(), tuple_extract_key_raw() and tuple_hash() as `static inline` wrappers for key_def->xxx().